app/views/auth/registrations/rules.html.haml

```diff
@@ -18,10 +18,11 @@
 
   %ol.rules-list
     - @rules.each do |rule|
+      - translation = rule.translations.find { |translation| translation.language == I18n.locale.to_s }
       %li
         %button{ type: 'button', aria: { expanded: 'false' } }
-          .rules-list__text= rule.text
-          .rules-list__hint= rule.hint
+          .rules-list__text= translation&.text || rule.text
+          .rules-list__hint= translation&.hint || rule.hint
```

Comment:
How about moving the translation lookup logic into a method on the `Rule` model?

It could also include the fallback logic and maybe even memoization. That could simplify this template a lot.

Reply:
I'm having a look and it's not super obvious how this could be done. I mean we could have a `rule.translated_text_for(I18n.locale.to_s)` but then memoizing the correct translation would be pretty awkward imho.

Reply:
I think that even without memoization it would be an improvement, so line 21 above could become something like:

```ruby
translation = rule.translation_for(I18n.locale)
```

And lines 24 and 25 could be made simpler.

Yes, memoizing when the result depends on a parameter is a bit awkward. I usually reach for a `Hash` and do something like:

```ruby
def translation_for(locale)
  @cached_translations ||= {}
  @cached_translations[locale] ||= (translations.find_by(language: locale) || self)
end
```

It is not pretty, but gets the job done.

Reply:
I went with something like that, thanks!